lib/takeaway.rb

```diff
@@ -0,0 +1,48 @@
+class Takeaway
+  attr_reader :basket
+
+  def initialize( restaurant, menu)
+    @menu = menu
+    @restaurant = restaurant
+    @basket = {}
+  end
+
+  def add( name, qty = 1)
+    fail "#{name} not available" unless @menu.contains?(name)
+    if @basket.has_key?(name) then
+      @basket[name] += qty
+    else
+      @basket[name] = qty
+    end
+  end
+
+  def remove ( name, qty = 1)
```

Comment:
Style/SpaceAfterMethodName: Do not put a space between a method name and the opening parenthesis. (https://github.com/bbatsov/ruby-style-guide#parens-no-spaces)